### PR TITLE
add parquet_token_v2_processor

### PR DESCRIPTION
**Pull request by yuunlimm**
### Description
Add parquet token processor that is responsible for 
~~1. collection_v2~~ this will be omitted for now. 
2. token_datas_v2
3. token_ownerships_v2

### Test Plan
token_ownerships
Screenshot 2024-07-15 at 2 08 25 PM
Screenshot 2024-07-15 at 2 08 23 PM
Screenshot 2024-07-15 at 2 08 25 PM

collection_v2
Screenshot 2024-07-15 at 2 08 05 PM

### CODE REVIEW

File: rust/processor/src/db/common/models/token_v2_models/parquet_v2_collections.rs [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+    pub description: String,
+    pub uri: String,
+    pub current_supply: String,
+    pub max_supply: Option<String>,
+    pub total_minted_v2: Option<String>,
+    pub mutable_description: Option<bool>,
+    pub mutable_uri: Option<bool>,
+    pub table_handle_v1: Option<String>,
+    pub token_standard: String,
+    #[allocative(skip)]
```
Comment (ying-w) [inline rust/processor/src/db/common/models/token_v2_models/parquet_v2_collections.rs:52]: we used to also have a `collection_properties` field

File: rust/processor/src/db/common/models/token_v2_models/parquet_v2_token_datas.rs [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+    pub write_set_change_index: i64,
+    pub token_data_id: String,
+    pub collection_id: String,
+    pub token_name: String,
+    pub largest_property_version_v1: Option<u64>,
+    pub token_uri: String,
+    pub token_properties: String,
+    pub description: String,
+    pub token_standard: String,
+    pub is_fungible_v2: Option<bool>,
```
Comment (ying-w) [inline rust/processor/src/db/common/models/token_v2_models/parquet_v2_token_datas.rs:42]: @rtso what's the diff between token_standard and is_fungible_v2?

**Review by ying-w — COMMENTED** [review]
Left some feedback for
- collections
- token_data
- token_ownerships

did you already move collection_data?

File: rust/processor/src/db/common/models/token_v2_models/parquet_v2_token_datas.rs (hunk 2 above)
Comment (rtso) [reply rust/processor/src/db/common/models/token_v2_models/parquet_v2_token_datas.rs:42]: `token_standard` is whether the token is v1 or v2. `is_fungible_v2` is whether this token is also fungible (it's an implementation detail based on the contract)

**Comment by yuunlimm** [comment]
> Left some feedback for
> 
> * collections
> * token_data
> * token_ownerships
> 
> did you already move collection_data?

collection_data wasn't a part of parquet. do we need in bq?

File: rust/processor/src/db/common/models/token_v2_models/parquet_v2_collections.rs (hunk 1 above)
Comment (yuunlimm) [reply rust/processor/src/db/common/models/token_v2_models/parquet_v2_collections.rs:52]: collections is not going to be migrated over in this phase. we can ignore this file for now

**Comment by yuunlimm** [comment]
Screenshot 2024-07-25 at 2 44 26 PM
Mainnet backfill is done.
@larry-aptos

File: rust/processor/src/db/common/models/token_v2_models/parquet_v2_token_ownerships.rs [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+        let non_transferrable_by_owner = !object_core.allow_ungated_transfer;
+
+        ownerships.push(Self {
+            txn_version: token_data.txn_version,
+            write_set_change_index: token_data.write_set_change_index,
+            token_data_id: token_data_id.clone(),
+            property_version_v1: 0,
+            owner_address: Some(owner_address.clone()),
+            storage_id: storage_id.clone(),
+            amount: "0".to_string(),
```
[the remark this answers is not in the document]
Comment (yuunlimm) [reply rust/processor/src/db/common/models/token_v2_models/parquet_v2_token_ownerships.rs:101]: sounds good. will also add a comment

**Review by larry-aptos — APPROVED** [review]

File: rust/processor/src/db/common/models/token_v2_models/parquet_v2_token_ownerships.rs [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+            // If it's a self transfer then skip
+            if transfer_event.get_to_address() == transfer_event.get_from_address() {
+                continue;
+            }
+            ownerships.push(Self {
+                txn_version: token_data.txn_version,
+                // set to negative of event index to avoid collison with write set index
+                write_set_change_index: -1 * event_index,
+                token_data_id: token_data_id.clone(),
+                property_version_v1: 0,
```
[the remark this answers is not in the document]
Comment (yuunlimm) [reply rust/processor/src/db/common/models/token_v2_models/parquet_v2_token_ownerships.rs:121]: not quite sure if it was added as a part of migration.

**Pull request merged**